packages/test/test-utils/src/testObjectProvider.ts

```diff
@@ -122,14 +122,63 @@ function getDocumentIdStrategy(type?: TestDriverTypes): IDocumentIdStrategy {
     }
 }
 
+export class ErrorTrackingLogger extends TelemetryLogger{
+    constructor(private readonly baseLogger: ITelemetryBaseLogger){
+        super();
+    }
+
+    private readonly expectedEvents: (ITelemetryGenericEvent | undefined)[] = []
+    private readonly unexpectedErrors: ITelemetryBaseEvent[] =[];
+
+    public registerExpectedEvent(... events: ITelemetryGenericEvent[]){
+        this.expectedEvents.push(... events);
+    }
+
+    send(event: ITelemetryBaseEvent): void {
+        const ee = this.expectedEvents[0];
+        if(ee?.eventName === event.eventName){
+            let matches = true;
+            for(const key of Object.keys(ee)){
+                if(ee[key] !== event[key]){
+                    matches = false;
+                    break;
+                }
+            }
+            if(matches){
+                this.expectedEvents.shift();
+                if(event.category === "error"){
+                    event.category = "generic";
+                }
+            }
+        }
+        if(event.category === "error"){
+            this.unexpectedErrors.push(event);
+        }
+
+        this.baseLogger.send(event)
+    }
+
+    public results(){
```

Comment:
It's surprising that getting the results clears the state. More importantly, if someone used this logger and didn't call `results`, the next usage would be messed up (I know, no one should use the logger directly so it's probably ok).  Can the clearing/resetting of this thing be moved to some `afterEach` somewhere?

Reply:
there is no after each at this layer. i can give this a better name